Millicast SDK Sample App in Swift/MediaSettingsView.swift

```diff
@@ -0,0 +1,114 @@
+//
+//  SettingsMediaView.swift
+//  SwiftSa iOS
+//
+//  Created by Szymczak, Marcin on 28/12/2023.
+//
+
+import Foundation
+
+
+import AVFoundation
+import SwiftUI
+import MillicastSDK
+
+/**
+ * UI for Millicast platform settings.
+ */
+struct MediaSettingsView: View {
+    
+    @ObservedObject var mcMan: MillicastManager
+    
+    init(manager mcMan: MillicastManager) {
+        self.mcMan = mcMan
+    }
+    
+    var body: some View {
+        VStack(alignment: .center, spacing: 0.0) {
+            VStack(){
+                HStack {
+                    Text("Media Capture Settings")
+                        .font(.title3)
+                    .fontWeight(.bold)
+                }.frame(
+                    minWidth: 0,
+                    maxWidth: .infinity,
+                    alignment: .center
+                  )
+                List {
+                    Picker("Video Source", selection: $mcMan.videoSourceIndex) {
+                        ForEach(mcMan.getWrappedVideoSourceList()){src in
+                            Text("\(src.source.getName()) [\(src.id)]").tag(src.id)
+                        }
+                    }
+                    .onChange(of: mcMan.videoSourceIndex){tag in mcMan.setVideoSourceIndex(tag, setCapIndex: true)}
```

Comment:
I read this code, and starting to understand how it's supposed to work. So, please tell me if this understanding is correct:
- there's a bunch of new types in `MCTypes.swift`, that map objects to integer identifiers,
- in `mcMan.getWrappedXYZ()` we get the list of entities from the SDK, and then create wrapped list, where the IDs are indices from the vector returned from the SDK
- the indices are used in the UI for selecting proper entity and calling back into the mcMan/SDK

Now, with lists like codecs, these are (reasonable assumption) immutable, each time you get them from the SDK, you get the same. Indices are a good ID.

With lists like video sources, these are highly volatile. The safety is achieved by ensuring that *only* in this code here, when constructing the UI, we invoke `getWrappedXYZ()` which is allowed to fetch the list from the SDK, and update it. mcManager caches audio sources and video sources, and only updates them when this View here is created.

Then, in let's say `setVideoSource`, there's a call to `getVideoSources(refresh:false)`, which may or may not refresh the cached list, but if hte cached list is not null, it won't refresh it, and will just return it; then there's a bunch of boundary checking of the index against list size and zero, and then there's `getVideoSources()` called again, this time not to query the size of the list, but to get the indexed element. This time, we're guaranteed that it won't refresh the cached list, becuase it's definitely not null; then the indexed source is stored in the `videoSource` member in the manager.

It seems to me that this is a code that works. Looks overly complicated, and looks very suspicious - at the first sight, you almost want to scream "this won't work!" - but it works. Still, can it be simplified? Especially, if the UI code can not handle object values assigned to the individual picker items, and must operate on indices, can the caching code be moved from mcMan to the view? (let the view get video source list and cache it, drop the idea of wrapped types, in the picker change use the picker value to index the vector of stored sources, do not pass video source index into mcMan, pass the MCVideoSource object?)

Reply:
Note that IMO you don't have to implement anything of the above right now, also, I know nothing about SwiftUI; but if some parts of what I said sound reasonable, please create a jira issue to follow up with a cleanup

Reply:
>here's a bunch of new types in MCTypes.swift, that map objects to integer identifiers

>in mcMan.getWrappedXYZ() we get the list of entities from the SDK, and then create wrapped list, where the IDs are indices from the vector returned from the SDK

>Especially, if the UI code can not handle object values assigned to the individual picker items, and must operate on indices

The UI `Picker` component in swift requires the objects to _"conform to the Identifiable protocol"_, which translates to "have a field named `id`",so it can handle object values, but the field named `id` must be present


>With lists like video sources, these are highly volatile.

In short, using the list indices for this is bad, I agree. The `MCManager` assumes sources are changed by Id, but doesn't actually provide any way to get id's unique to the given source, which IMO could be refactored.
The SDK is supposed  to expose some kind of id according to [docs](https://millicast.github.io/doc/cpp/classmillicast_1_1_source.html#a285f3371dde0e3786910cd758c903531) but for some reason the ObjC SDK just doesn't :/
However, The SDK provides a getter for a `unique_id` of a source, so perhaps this can be used as an identifier to. replace the indices in the whole app

>can the caching code be moved from mcMan to the view

The same cache is used in the `PublisherView`, which also allows the user to change sources (a bit simpler, can only cycle through the list one by one), so I think it has to stay in the manager (or each view would need to cache it separately)

Reply:
Well, the cache is still *only* there for the purpose of being used to implement the UI, it does not belong in the model-like classes like mcManager. Perhaps then implementing some helper class to be used in viewer and publisher views, which would allow creating and manipulating cache easily, reusing the code, would be a good solution? anyway, that is for the future considerations.

> However, The SDK provides a getter for a unique_id of a source, so perhaps this can be used as an identifier to. replace the indices in the whole app

That may be a very good idea.